metadata-service/factories/src/main/java/com/linkedin/gms/factory/kafka/ThreadPoolContainerCustomizer.java

```diff
@@ -0,0 +1,19 @@
+package com.linkedin.gms.factory.kafka;
+
+import org.apache.avro.generic.GenericRecord;
+import org.springframework.kafka.config.ContainerCustomizer;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+
+public class ThreadPoolContainerCustomizer
```

Comment:
Will this have any processing throughput implications in the default case?

Reply:
It's possible, it might make sense to performance test this before pushing it through. Logically it shouldn't, it's essentially doing the same thing as before just with thread reuse.